### PR TITLE
internal: add workflow to check for outdated integrations

### PR DESCRIPTION
**Pull request by quinna-h**
### What does this PR do?

Duplicate of #2936 on the v2-dev branch, following the policy that we need a duplicate PR on the `v2-dev` branch before GA release.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.
- [ ] For internal contributors, a matching PR should be created to the `v2-dev` branch and reviewed by @DataDog/apm-go.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: integration_go.mod [hunk 1]
```diff
@@ -0,0 +1,91 @@
+module gopkg.in/DataDog/dd-trace-go.v1 // This file is manually maintained and updated
```
Comment (hannahkm) [inline integration_go.mod:1]: we should probably change this module name to be `github.com/DataDog/dd-trace-go/v2` to match the rest of the mod files (see the `go.mod` file in this directory).

File: .github/workflows/apps/latest_major_version.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
```
Comment (hannahkm) [inline .github/workflows/apps/latest_major_version.go:4]: nit: 2024?

File: integration_go.mod [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+go 1.22.0
```
Comment (hannahkm) [inline integration_go.mod:3]: Could we run a `go mod tidy` in this directory? There's some contribs that we've removed support for in v2 that are still in here, so I think a tidy might help take care of those.

Or, if this is manually updated, let's just take a look at the `contrib` folder and see what's still there and what's not. If you'd like to connect on that, feel free to find me!

**Review by hannahkm — CHANGES_REQUESTED** [review]

File: integration_go.mod (hunk 3 above)
Comment (quinna-h) [reply integration_go.mod:3]: done!

**Review by hannahkm — APPROVED** [review]

**Pull request merged**